File: main-svc/src/main/java/com/chilly/main_svc/model/Place.java
package com.chilly.main_svc.model;

import com.chilly.main_svc.mapper.ListToStringConverter;
import jakarta.persistence.*;
import lombok.*;

import java.util.LinkedHashSet;
import java.util.List;
import java.util.Set;

@Entity(name = "place")
@Table(name = "places")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Place {
    @Id
    @Column(name = "id")
    private Long id;

    @Column(name = "name")
    private String name;

    @Column(name = "address")
    private String address;

    @Column(name = "website")
    private String website;

    @Column(name = "y_page")
    private String yPage;

    @Column(name = "rating")
    private Double rating;

    @Column(name = "images")
    @Convert(converter = ListToStringConverter.class)
    private List<String> images;

    @Column(name = "phone")
    private String phone;

    @Column(name = "social")
    @Convert(converter = ListToStringConverter.class)
    private List<String> social;

    @Column(name = "open_hours")
    @Convert(converter = ListToStringConverter.class)
    private List<String> openHours;

    @OneToMany(mappedBy = "place", orphanRemoval = true)
    @Builder.Default
    private Set<Visit> visits = new LinkedHashSet<>();

}

File: places-svc/src/main/kotlin/com/chilly/places_svc/model/Place.kt
package com.chilly.places_svc.model

import com.chilly.places_svc.mapper.ListToStringConverter
import jakarta.persistence.*

@Entity(name = "place")
@Table(name = "places")
class Place(
    @Id
    @Column(name = "id", nullable = false)
    val id: Long,

    @Column(name = "name", nullable = false)
    var name: String,

    @Column(name = "address", nullable = true)
    var address: String? = null,

    @Column(name = "website", nullable = true)
    var website: String? = null,

    @Column(name = "latitude", nullable = false)
    var latitude: Double = 0.0,

    @Column(name = "longitude", nullable = false)
    var longitude: Double = 0.0,

    @Column(name = "y_page", nullable = true)
    var yPage: String? = null,

    @Column(name = "rating", nullable = true)
    var rating: Double? = null,

    @Column(name = "images", nullable = true)
    @Convert(converter = ListToStringConverter::class)
    var images: List<String>? = null,

    @Column(name = "phone", nullable = true)
    var phone: String? = null,

    @Column(name = "social", nullable = true)
    @Convert(converter = ListToStringConverter::class)
    var social: List<String>? = null,

    @Column(name = "open_hours", nullable = true)
    @Convert(converter = ListToStringConverter::class)
    var openHours: List<String>? = null,
) {

    @OneToMany(mappedBy = "place", orphanRemoval = true)
    var visits: MutableSet<Visit> = mutableSetOf()
}